### ISSUE TITLE
Add compatibility for commonjs

### ISSUE THREAD

**Issue by jaszhix** [linuxmint/cjs#32]
I am wanting to use Lodash with my applet, but I am finding it difficult since the CJS environment doesn't have a browser or node environment. This makes it difficult to develop any Cinnamon extensions with a modern toolchain. 

I could probably make a version of Lodash CJS compatible by taking it out of its IIFE among other changes, but in general I think that's too tedious. I'm not sure if this project intends to diverge from GJS too much, but I think it would give the Cinnamon ecosystem a unique benefit.

**Comment by donnisnoni95** [comment]
try seed https://github.com/GNOME/seed

**Comment by icarter09** [comment]
@jaszhix are you still looking to use Lodash or have you decided to try out another JS framework instead? I completely agree that moving all the IIFEs out would be very tedious.

**Comment by jaszhix** [comment]
@icarter09 I used a custom-modified version of Lodash for a while in Icing Task Manager, but as we upgraded the JS engine from mozjs24 to 38, most of the functions I was using with it could be replaced with native ones.

I ended up adding some commonjs-esque support to Cinnamon in a refactoring of the extension system because extension reloading broke on the switch to mozjs52. Its still not full support, but it can allow easier compatibility with some third party libraries without modifying them, like in the [calculator applet](https://github.com/linuxmint/cinnamon-spices-applets/commit/73adbf7926f54a988b547a60f7672d109cb3e6ac). The main thing that needs to be improved is sub-directories. For a while I thought we would [move back](https://github.com/linuxmint/Cinnamon/pull/7349) to the CJS importer, and [fixed up](https://github.com/linuxmint/cinnamon-spices-applets/pull/1700) system-monitor@pixunil based on that branch. Then that revert PR turned out to have issues, and I ran into the sub-directory issue trying to fix the applet for the current Cinnamon importer. Since its the only applet structured that way, I haven't looked into it too much further.

This issue can be closed, since we have a foundation for some compatibility in Cinnamon now.
